apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/api/CompilerParams.kt

```diff
@@ -154,4 +154,10 @@ interface CompilerParams {
    * Default value: false
    */
   val generateAsInternal: Property<Boolean>
+
+  /**
+   * Whether to generate GraphQL enum types by matching as Kotlin sealed classes if they match one of provided regex patterns.
+   * By default any GraphQL enum type is generated as `enum class`.
+   */
+  val generateEnumAsSealedClass: ListProperty<String>
```

Comment:
Nit: Maybe name this `enumAsSealedClassPatterns`? `generate` is used above for Boolean properties and making the property plural would also hint that it is a List.

Reply:
Yeah, hard to come up with proper name my thought `enumAsSealedClassPatternFilters`   is ok for internal usage but was not great for public API. 
`generateSealedClassForEnums`, `sealedClassForEnums`, `enumsAsClasses`, `enumAsSealedClassPatterns`?

Or do you think `enumAsSealedClassPatternFilters` should be fine?

Reply:
I also think that the one used in the code internally is not super clear for external facing extension. We can make it plural easily to indicate that. Since they're `enum` in GraphQL convention, I like starting with `generateEnums`

What about

```
generateEnumsAsSealedClassesFor
```

or 

```
generateSealedClassesForEnums
```

Reply:
Maybe something that describes the end functionality instead of the codegen details ? 

```
exposeRawValueForUnknownEnums
```
or
```
exposeRawValueForEnumsMatching
```

Reply:
but then it's not clear that the end result is going to be a sealed class generated instead of regular enum, I think it's important to have `sealedClass` in there.
1 - `enumsAsSealedClassesMatching`
2 - `enumsAsClassesMatching`
3 - `sealedClassesForEnums`
4 - `sealedClassesForEnumsMatching`

Another option if this make sense make it lambda:
```
apollo {
  enumsAsSealedClasses {
    matching = [".*"]
  }
}
```

Reply:
I like `sealedClassesForEnumsMatching`

Reply:
I definitely like that too. 👍

Reply:
We got a winner 🎉 😃